---
layout: post
title:  Is it a fight if the other person doesn't know they are fighting
categories: [non-mass spec-based proteomics, future of proteomics, pore-based proteomics, aptamer-based proteomics]
excerpt: A constant discussion on the twitters, and in-person, is the rise of non-mass spec based proteomic techniques. This discussion is developing quite well from the mass spec side, except for one crucial point: we seem to think it is a point up for debate.
---

A constant discussion on the twitters and in-person meetings, is the rise of non-mass spec based proteomic techniques. This discussion is developing quite well on the mass spec side, except for one crucial point: we seem to think it is up for debate.

&nbsp;  

Some key premises, which contain the root of the argument:
* cost isn't actually important, but ease of data use is ([my response](https://twitter.com/neely615/status/1530653991801077763?s=20&t=k118iFtYQMzFiN_dZOoDBQ))
* if previous (and current) mass spec-based proteomics results that ignored proteoforms and PTMs were useful, then why not these newer non-MS techniques
* mass spec-based proteomics users aren't the target audience (it doesn't need to be better than mass spec to have widespread adoption)
* genomics cores can easily buy turn-key solutions (aptamer, pore-based, or techniques using existing sequencers) to "do" proteomics (ex. (Genomics Proteomics Core](https://www.bidmc.org/research/core-facilities/genomics-proteomics-core))
* non-mass spec-based techniques don't actually need to compete with mass spec capabilities (PTMs, proteoforms, etc.)
 - We get it, they can do everything and are awesome. I completely agree.

&nbsp;  

The big tweet "How to make MS-based proteomics accessible as DNA/RNA"?
<blockquote class="twitter-tweet"><p lang="en" dir="ltr">What will it take to make mass-spec proteomics as accessible as DNA/RNA sequencing currently is ? <a href="https://t.co/IhacovSa44">https://t.co/IhacovSa44</a></p>&mdash; Prof. Nikolai Slavov (@slavov_n) <a href="https://twitter.com/slavov_n/status/1530517564115255296?ref_src=twsrc%5Etfw">May 28, 2022</a></blockquote> <script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script> 


I am solidly in the "need a black box that reaches more biologists all over the world" camp (see [2005 Yates editorial](https://www.the-scientist.com/vision/proteomics-power-to-the-people-49117) too). Let mass spectrometers be machines not instruments.

&nbsp;  

This thread definitely argues this point of black box or not, but then drifts into the current supply of people with hardware backgrounds (you can unroll the thread from [this deep comment](https://twitter.com/bundyba1972/status/1532102338990772224?s=20&t=W_13_TpXrAK7Vn0H61CrJg)). An elusive reboot of the linear ion trap may come to our rescue, or certain MS1 or in-source fragmentation methods will make single quads viable for more applications, or maybe TOFs continue to make strides in the right direction. Point being, in my opinion, there is a need to do away with the need to be an expert at chemistry, molecular bio, electronics, physics, statistics, bioinformatics, etc. to run a mass spec and interpret data. It is unreasonable to ask a biologists to do all of this too. We need black boxes (for better or worse) or mass spec-based proteomics will never be accessible. Maybe then we can focus on the biology (and better experimental design).

&nbsp;  

A follow-up, kind of, was more specific to the pore-based proteomics advancements:
<blockquote class="twitter-tweet"><p lang="en" dir="ltr">Have you wondered what people outside of the teammassspec bubble think is the future of proteomics? Spoiler: itâ€™s not mass spec. Wild to hear.</p>&mdash; Ben Neely ðŸ‡ºðŸ‡¦ (@neely615) <a href="https://twitter.com/neely615/status/1532478656299294730?ref_src=twsrc%5Etfw">June 2, 2022</a></blockquote> <script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script> 

&nbsp;  

To be clear, I am not worried about this disruptive technology since I don't study mass specs, I explore biology, and want something to make that easier, whatever it may be. When I see ONT (moto: "Our goal is to enable the analysis of anything, by anyone, anywhere.") and people posting pictures of them analyzing genomes in their car in traffic, and then I look at the AutÃ³dromo Internacional do Algarve really fast, but it is super expensive, breaks quite a bit, and requires a support team to use. If my goal is to drive, an F1 may not be for me, and arguments about how well my Camry performs at 200 mph won't convince me to choose an F1. Still, for much of the work I am currently interested in, broadly under the umbrella of protein forensics, a mass spec is great at pattern recognition and has a definite advantage imo. But maybe even this is just my bias coming out.

&nbsp;  

Again, we think there is a dilemma because we are looking at the problem from within our bubble, and the bubble isn't even that big.


&nbsp;  
&nbsp;  
&nbsp;  
